demos/DemoSettings.tsx

```diff
@@ -178,20 +178,20 @@ export default function DemoSettings(props) {
               onSetFile={async (file) => {
                 setUploading(true);
                 if (Utilities.isEmpty(props.sessionKey)) {
-                  props.onSetModal({ name: 'ERROR', message: 'not authenticated' });
+                  props.onError('not authenticated');
```

Comment:
This lack of flexibility doesn't seem like an improvement at all. What if I wanted to render a warning, or another type of modal? Would I just keep adding `onError` `onWarning` `onNotice` `onOnboarding`

Reply:
i would argue that the previous version was a different type of inflexible. what if i wanted to change error reporting behavior so any messages would also get logged to the console, or transmitted to an analytics server? or what if i wanted to upgrade to a stacking toast system in the future? i would have to go and modify all the call sites to show a toast instead of a modal, or to add whatever extra behavior i wanted after each call site. to me, the intent of the code is to display a message, and the form of that message just happens to be a modal right now.

you're right that the code doesn't currently support different message levels right now. but if we want to support different levels of messages, we could add an optional second param with a level enum (or just an options struct). this is not even a breaking change aside from an automatic naming refactor, since it can be an optional parameter. that's why i think this is a better solution.

Reply:
Regarding reporting to an analytics server or adding functionality to a modal, I’m glad we can combine them at the callsite instead of within the modal itself. However, that’s not what I’m referring to. The previous `onSetModal` method was intended to allow the user to render a different modal, and the modal system should focus solely on managing that behavior—detecting the change and handling it correctly.

It’s fine to wrap `onSetModal` in something like `onError` for simplicity, but I do wonder if the additional layer of indirection is necessary, especially if the function’s name differs to suit the call site context.

The specific scenario I’m referring to doesn’t exist yet, so we can revisit this when it comes up.

Reply:
if some other type of modal is to be shown, what is the reason not to just `useModal()` in this component, instead of passing a callback prop with the same value?

Reply:
See our other conversation